lte/gateway/c/core/oai/tasks/s1ap/s1ap_mme_handlers.cpp

```diff
@@ -666,47 +655,51 @@ status_code_e s1ap_mme_handle_s1_setup_request(s1ap_state_t* state,
   }
 
   S1ap_SupportedTAs_t* ta_list = &ie_supported_tas->value.choice.SupportedTAs;
-  supported_ta_list_t* supp_ta_list = &enb_association->supported_ta_list;
-  supp_ta_list->list_count = ta_list->list.count;
+  magma::lte::oai::SupportedTaList* supported_ta_list_proto =
+      enb_association->mutable_supported_ta_list();
+  supported_ta_list_proto->set_list_count(ta_list->list.count);
 
   /* Storing supported TAI lists received in S1 SETUP REQUEST message */
-  for (int tai_idx = 0; tai_idx < supp_ta_list->list_count; tai_idx++) {
+  for (int tai_idx = 0; tai_idx < supported_ta_list_proto->list_count();
+       tai_idx++) {
     S1ap_SupportedTAs_Item_t* tai = NULL;
+    uint32_t tac;
+    magma::lte::oai::SupportedTaiItems* supported_tai_item =
+        supported_ta_list_proto->add_supported_tai_items();
     tai = ta_list->list.array[tai_idx];
-    OCTET_STRING_TO_TAC(&tai->tAC,
-                        supp_ta_list->supported_tai_items[tai_idx].tac);
-
+    OCTET_STRING_TO_TAC(&tai->tAC, tac);
+    supported_tai_item->set_tac(tac);
     bplmn_list_count = tai->broadcastPLMNs.list.count;
     if (bplmn_list_count > S1AP_MAX_BROADCAST_PLMNS) {
       OAILOG_ERROR(LOG_S1AP,
                    "Maximum Broadcast PLMN list count exceeded, count = %d\n",
                    bplmn_list_count);
     }
-    supp_ta_list->supported_tai_items[tai_idx].bplmnlist_count =
-        bplmn_list_count;
+    supported_tai_item->set_bplmnlist_count(bplmn_list_count);
     for (int plmn_idx = 0; plmn_idx < bplmn_list_count; plmn_idx++) {
-      TBCD_TO_PLMN_T(
-          tai->broadcastPLMNs.list.array[plmn_idx],
-          &supp_ta_list->supported_tai_items[tai_idx].bplmns[plmn_idx]);
+      char plmn_array[PLMN_BYTES] = {0};
+      plmn_t plmn = {0};
```

Comment:
Can you remove the `0` between the brackets because it initialises only the first field in the struct `plmn_t`.